lib/matplotlib/dates.py

```diff
@@ -1314,11 +1332,34 @@ def get_locator(self, dmin, dmax):
         #  bysecond, unused (for microseconds)]
         byranges = [None, 1, 1, 0, 0, 0, None]
 
+        # Required attributes to get width from figure's normal points
```

Comment:
I think this should all be wrapped in `if spacing:`

I don't get why you are using getattr for all these.  If these attributes don't exist, you might as well error here because you weren't passed an axis...

Reply:
If we raise exceptions it would break our existing mock class tests that might have aaxis attributes as none.

Reply:
Which tests?  I don't understand what a test would be doing down in here w/o the axes being set (for instance).

Reply:
OK, well I'd just check if whichever one of these is None and set spacing=False if that happens